.github/workflows/build-deploy-microservices.yml

```diff
@@ -0,0 +1,246 @@
+      uses: actions/upload-artifact@v2
+      with:
+        name: gateway.war
+        path: gateway/build/libs/gateway-*.war
+    - name: Upload User Management Microservice Artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: user-management.war
+        path: user-management/build/libs/user-management-*.war
+#    - name: Upload Release Artifact
+#      if: github.event_name == 'release' && github.event.action == 'created'
+#      uses: actions/upload-release-asset@v1
+#      env:
+#        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+#      with:
+#        upload_url: ${{ github.event.release.upload_url }}
+#        asset_path: build/libs/Artemis-${{ github.event.release.tag_name }}.war
+#        asset_name: Artemis.war
+#        asset_content_type: application/x-webarchive
+
+  docker_push_images:
+    needs: build
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+        - image_name: "artemis-service"
+          dockerfile_path: "src/main/docker"
+        - image_name: "artemis-gateway"
+          dockerfile_path: "gateway/src/main/docker"
+        - image_name: "artemis-usermanagement-service"
+          dockerfile_path: "user-management/src/main/docker"
+    steps:
+    - name: Download Artifact
+      uses: actions/download-artifact@v2
+      with:
+        name: Artemis.war
+    - name: Download Dockerfile
+      run: curl -O ${{ env.RAW_URL }}/${{ matrix.dockerfile_path }}/Dockerfile -O ${{ env.RAW_URL }}/${{ matrix.dockerfile_path }}/entrypoint.sh
+    - name: Build Docker Image
+      run: docker build --build-arg WAR_PATH=. -t ${{ matrix.image_name }} .
+
+    - name: Compute Tag
+      uses: actions/github-script@v5
+      id: compute-tag
+      with:
+        result-encoding: string
+        script: |
+          if (context.eventName === "pull_request") {
+            return "pr-" + context.issue.number;
+          }
+          if (context.eventName === "push") {
+            if (context.ref.startsWith("refs/tags/")) {
+              return context.ref.slice(10);
+            }
+            if (context.ref === "refs/heads/develop") {
+              return "latest";
+            }
+          }
+          return "FALSE";
+
+    # Push to GitHub Container Registry
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+      if: ${{ steps.compute-tag.outputs.result != 'FALSE' }}
+    - name: Push to GitHub Container Registry
+      env:
+        TAG: ${{ steps.compute-tag.outputs.result }}
+      run: |
+        docker tag ${{ matrix.image_name }} ghcr.io/ls1intum/${{ matrix.image_name }}:$TAG
+        docker push ghcr.io/ls1intum/${{ matrix.image_name }}:$TAG
+      if: ${{ steps.compute-tag.outputs.result != 'FALSE' }}
+
+    # TODO: Push to Docker Hub (develop + tag)
+
+    # TODO: Push to Chair Habour (??)
+
+  deploy:
```

Comment:
I think you should remove the deploy stage completely. The Deployment described here is not related to the micro-services we like to test/deploy with this workflow.